### ISSUE TITLE
handling in case of status code 204

### ISSUE THREAD

**Issue by SiddiquiImran** [pagekit/vue-resource#600]
### Reproduction Link

await Post('my method name', JSON.stringify(my param))
        .then(response => {
          console.log('response: ', response)
          if (Helper.IsValid(response, true)) {
                 // code need to fire
          }
        }, response => {
          console.log('error response: ', response)
        })

### Steps to reproduce
Chrome 57 / Windows 10 Pro

### What is Expected?
In case of http return code 204 it should fire success handler of vue-resource.

### What is actually happening?
In case of http return code 204 it should fire success handler but it's firing error handler of vue-resource..

**Comment by ivanwitzke** [comment]
couldn't you just check it with ```response.ok```?

**Comment by KyleNeedham** [comment]
I'm also having trouble with the `201` status code being rejected. This is particularly difficult to handle when using a `try catch`  statement. Anybody why it would be rejected?

```javascript
try {
  this.hasPendingRequest();
  const request = await this.$http.post(uri, data);

  // handle good response
} catch (e) {
  // handle bad response
} finally {
  this.clearPendingRequest();
}
```

**Comment by SiddiquiImran** [comment]
Hi @ivanwitzke ,
no matter it always goes to error section so you can not check it with "response.ok", for the time being i have changed it with "axios" plugin.

**Comment by herrmannplatz** [comment]
@KyleNeedham not sure if 201 is the root cause of the rejection.

Made a small example which seems to work fine.

https://jsfiddle.net/k9w9nrv3/

**Comment by SiddiquiImran** [comment]
Hi @herrmannplatz ,

Service in the jsfiddle returning 403
![image](https://user-images.githubusercontent.com/20870175/29600214-429141c8-87f2-11e7-9039-6d9b8b27910b.png)

**Comment by herrmannplatz** [comment]
Not sure why you get a 403. You might wanna check the error message in detail. 

<img width="973" alt="bildschirmfoto 2017-08-23 um 09 20 55" src="https://user-images.githubusercontent.com/174685/29603738-96a14f04-87e5-11e7-8873-19b07d4a42b3.png">

**Comment by SiddiquiImran** [comment]
May be issue when using with vue application :)

**Comment by KyleNeedham** [comment]
@herrmannplatz You're right I just updated the endpoint to use a `201` again and it works fine. Must have been something else causing the error, also tried a `204` and that worked as well.
